### DISCUSSION TITLE
Issue with Manual Install

### DISCUSSION THREAD

**Discussion by ghostlyinfluence** [Xilinx/RapidWright#423] [Q&A]
Hello,

I am not very familiar with Java / Gradle and I am having an issue with the manual install.

I have:
1. Cloned the repository and downloaded and extracted the jar and data folders and placed them inside the repository folder.
2. Added $RAPIDWRIGHT_PATH and CLASSPATH to my .bashrc

Then, when I go to build the project via gradle "gradle build -p $RAPIDWRIGHT_PATH" I get the following error:

FAILURE: Build failed with an exception.

* Where:
Build file '/mnt/d/Development/rwdev/RapidWright/build.gradle' line: 182

* What went wrong:
Could not compile build file '/mnt/d/Development/rwdev/RapidWright/build.gradle'.
> startup failed:
  build file '/mnt/d/Development/rwdev/RapidWright/build.gradle': 182: unexpected token: -> @ line 182, column 83.
     ludePatterns.stream().map((p) -> 'python
                                   ^

  1 error

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s

Help would be appreciated!

**Comment by clavin-xlnx** [comment]
Sorry you've run into this issue. We have made several changes and improvements to the Gradle build setup in RapidWright since the [Manual Install](http://www.rapidwright.io/docs/Manual_Install.html) page was written.  Try these commands instead:
```
git clone https://github.com/Xilinx/RapidWright.git
cd RapidWright
./gradlew compileJava
java -cp bin:jars/* com.xilinx.rapidwright.device.browser.DeviceBrowser
```
I will update the documentation soon and will update this thread when I do.  Let me know if you run into any issues with these commands.

**Reply by clavin-xlnx** [reply]
Just as an update, we've revised the Manual Install page here: http://www.rapidwright.io/docs/Manual_Install.html#manual-install.

**Comment by ghostlyinfluence** [comment]
Chris,

Perfect and thank you for your help!

On Tue, May 17, 2022 at 11:51 PM Chris Lavin ***@***.***>
wrote:

> Just as an update, we've revised the Manual Install page here:
> http://www.rapidwright.io/docs/Manual_Install.html#manual-install.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Xilinx/RapidWright/discussions/423#discussioncomment-2772278>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMWZ35SIDXEUDUDWNDYSC6DVKRSNZANCNFSM5V7OYIBQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
Colin Burgin
